Source/WebCore/page/Frame.cpp

```diff
@@ -209,6 +210,35 @@ Ref<NavigationScheduler> Frame::protectedNavigationScheduler() const
     return m_navigationScheduler.get();
 }
 
+std::optional<size_t> Frame::indexInFrameTreeSiblings() const
+{
+    if (!tree().parent())
+        return std::nullopt;
+
+    for (size_t i = 0; i < tree().parent()->tree().childCount(); i++) {
+        if (auto child = tree().parent()->tree().child(i); child->frameID() == this->frameID())
+            return i;
+    }
+
+    ASSERT_NOT_REACHED("This frame should be in its own tree");
+    return 0;
```

Comment:
This should indeed never happen, but I think it would be slightly better to return nullopt here.